File: Linkpoint/src/main/java/com/linkpoint/slproto/handler/SLEventQueueMessageHandler.java
package com.linkpoint.slproto.handler;

import com.linkpoint.slproto.caps.SLCapEventQueue;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target({ElementType.METHOD})
@Retention(RetentionPolicy.RUNTIME)
public @interface SLEventQueueMessageHandler {
    SLCapEventQueue.CapsEventType eventName();
}

File: Linkpoint/src/main/java/com/linkpoint/slproto/handler/SLEventQueueMessageHandler.kt
package com.linkpoint.slproto.handler

import com.linkpoint.slproto.caps.SLCapEventQueue

@Target(AnnotationTarget.FUNCTION)
@Retention(AnnotationRetention.RUNTIME)
annotation class SLEventQueueMessageHandler(
    val eventName: SLCapEventQueue.CapsEventType
)